CMakeLists.txt

```diff
@@ -254,131 +272,71 @@ endif()
 #   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
 #   BOOST_ROOT - root of Boost installation
 #   BOOST_INCLUDEDIR - include directory for Boost
+#   CEREAL_INCLUDE_DIR - include directory for cereal
 #   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
 #   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
 #   MATHJAX_ROOT - root of MathJax installation
-find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
 
+if(CMAKE_CROSSCOMPILING)
+  get_deps(https://github.com/xianyi/OpenBLAS/releases/download/v0.3.13/OpenBLAS-0.3.13.tar.gz OpenBLAS OpenBLAS-0.3.13.tar.gz)
+  if(NOT MSVC)
+    execute_process(COMMAND make TARGET=ARMV8 BINARY=64 HOSTCC=gcc CC=${CMAKE_C_COMPILER} FC=${CMAKE_FORTRAN_COMPILER} NO_SHARED=1
+                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13)
+    file(GLOB OPENBLAS_LIBRARIES "${CMAKE_BINARY_DIR}/deps/OpenBLAS-0.3.13/libopenblas.a")
+    # set(OPENBLAS_LIBRARIES ${OPENBLAS})
+  endif()
+  find_library(GFORTRAN NAMES libgfortran.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  find_library(PTHREAD NAMES libpthread.a PATHS ${CMAKE_SYSROOT}/usr/lib)
+  set(MLPACK_LIBRARIES ${MLPACK_LIBRARIES} ${OPENBLAS_LIBRARIES} ${GFORTRAN} ${PTHREAD})
+endif()
+
+# Remove support for older 32bit system, if we need to have 32bit system
+# Since we are cross compiling, We will download it into the deps folder,
+if(DISABLE_DOWNLOADS)
+  find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
+elseif(NOT ARMADILLO_FOUND AND (NOT DISABLE_DOWNLOADS))
+  get_deps(http://files.mlpack.org/armadillo-10.3.0.tar.gz armadillo armadillo-10.3.0.tar.gz)
```

Comment:
Not sure but maybe it makes sense to serve `armadillo-latest` on the files server, that way we don't have to update the CMake file everytime there is a new armadillo release.

Reply:
I guess if we go for it we should do it for the other packages as well, OpenBLAS, cereal.

Reply:
Yeah, agreed, but this will add more work to do on the mlpack.org side for maintenance.
Also, cereal development seems to be slow, so it might work if we add `cereal-latest`
It might be a good idea to do this :+1:

Reply:
@rcurtin @zoq if you agree with this solution, I am happy to open a pull request on mlpack.org and add armadillo, cereal, and openblas. Let me know what do you think